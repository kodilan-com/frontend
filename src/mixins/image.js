const emptyLogo = require('../assets/images/empty-company-logo.png');

export default {
  methods: {
    handleImageError() {
      if (this.$refs.logo) {
        this.$refs.logo.src = emptyLogo;
      }
    },
  },
};
