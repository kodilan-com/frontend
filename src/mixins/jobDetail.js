import formatDate from '../utils/date';
import { JOB_TYPE_MAP } from '../store/constants';

export default {
  props: {
    post: {
      type: Object,
      required: true,
    },
  },
  computed: {
    detailsUrl() {
      return `/ilanlar/${this.post.slug}`;
    },
    company() {
      return this.post.company;
    },
    timeago() {
      return formatDate(this.post.updated_at);
    },
    jobType() {
      return JOB_TYPE_MAP[this.post.type] || '';
    },
    postTypeClassName() {
      const CLASS_MAP = {
        0: 'full-time',
        1: 'full-time',
        2: 'part-time',
        3: 'internship',
        4: 'freelance',
      };

      return {
        [CLASS_MAP[this.post.type]]: true,
      };
    },
  },
};
