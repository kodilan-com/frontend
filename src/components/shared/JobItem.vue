<script>
import JobTypeBadge from './JobTypeBadge';
import jobDetail from '../../mixins/jobDetail';
import image from '../../mixins/image';

export default {
  mixins: [jobDetail, image],
  components: {
    JobTypeBadge,
  },
};
</script>

<template>
  <router-link
    :to="detailsUrl"
    :class="postTypeClassName"
    class="listing tag-post-link"
  >
    <div class="listing-logo tag-post-link">
      <img
        :src="company.logo"
        :alt="company.name"
        ref="logo"
        loading="lazy"
        @error="handleImageError"
        class="tag-post-link"
      >
    </div>
    <div class="listing-title tag-post-link">
      <h4>
        <span class="title tag-post-link">
          {{ post.position }}
        </span>
        <job-type-badge :post="post" />
        <span class="timeago wide">
          {{ timeago }}
        </span>
      </h4>
      <ul class="listing-icons">
        <li>
          <i class="ln ln-icon-Management" />
          <router-link :to="`/firmalar/${company.slug}`">
            {{ company.name }}
          </router-link>
        </li>
        <li>
          <i class="ln ln-icon-Map2" />
          <router-link :to="`/ilan-ara?location=${post.location}`">
            {{ post.location }}
          </router-link>
        </li>
        <li class="timeago-item">
          <i class="ln ln-icon-Clock" />
          <span>{{ timeago }}</span>
        </li>
        <li>
          <div
            v-for="tag in post.tags"
            :key="tag.slug"
            class="listing-date post-tag tag-post-tag"
          >
            <router-link :to="`/etiket/${tag.slug}`" class="tag-post-tag">
              {{ tag.name }}
            </router-link>
          </div>
        </li>
      </ul>
    </div>
  </router-link>
</template>

<style lang="scss">
.listing {
  img {
    width: 60px;
    height: 60px;
  }

  .post-tag {
    margin-right: 5px;
  }

  .listing-type {
    top: 30px;
  }

  .timeago-item {
    display: none;

    span {
      color: #26ae61;
    }
  }

  .timeago.wide {
    font-size: 12px;
    color: #808080;
    position: absolute;
    right: 25px;
    top: 58px;
  }
}
</style>
