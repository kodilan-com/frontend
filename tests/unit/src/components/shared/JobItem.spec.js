import { shallowMount } from '@vue/test-utils';
import JobItem from '@/components/shared/JobItem';
import jobDetail from '@/mixins/jobDetail';
import image from '@/mixins/image';

jest.mock('@/mixins/jobDetail');
jest.mock('@/mixins/image');

describe('JobItem.vue', () => {
  let post;
  let company;
  let timeago;
  let wrapper;

  beforeEach(() => {
    post = {
      position: 'TEST_POSITION',
      location: 'TEST_LOCATION',
      tags: [
        { slug: 'TEST_SLUG_1', name: 'TEST_SLUG_NAME_1' },
        { slug: 'TEST_SLUG_2', name: 'TEST_SLUG_NAME_2' },
      ],
    };
    company = {
      logo: 'TEST_LOGO',
      name: 'TEST_NAME',
      slug: 'TEST_SLUG',
    };
    timeago = 'TEST_TIME_AGO';

    jobDetail.computed.company.mockReturnValue(company);
    jobDetail.computed.postTypeClassName.mockReturnValue('TEST_CLASS_NAME');
    jobDetail.computed.timeago.mockReturnValue(timeago);

    image.methods.handleImageError = jest.fn();

    wrapper = shallowMount(JobItem, {
      stubs: ['router-link'],
      propsData: {
        post,
      },
    });
  });

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should execute handleImageError if any error occurs while loading image', () => {
    const imageElement = wrapper.find('img');

    imageElement.trigger('error');

    expect(image.methods.handleImageError).toHaveBeenCalled();
  });

  it('should pass post to JobTypeBadge component', () => {
    const jobTypeBadge = wrapper.find('job-type-badge-stub');

    expect(jobTypeBadge.props('post')).toBe(post);
  });
});
