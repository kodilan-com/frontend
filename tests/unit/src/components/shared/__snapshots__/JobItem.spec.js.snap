// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem.vue should match snapshot 1`] = `
<a
  class="listing tag-post-link TEST_CLASS_NAME"
>
  <div
    class="listing-logo tag-post-link"
  >
    <img
      alt="TEST_NAME"
      class="tag-post-link"
      loading="lazy"
      src="TEST_LOGO"
    />
  </div>
   
  <div
    class="listing-title tag-post-link"
  >
    <h4>
      <span
        class="title tag-post-link"
      >
        
        TEST_POSITION
      
      </span>
       
      <job-type-badge-stub
        post="[object Object]"
      />
       
      <span
        class="timeago wide"
      >
        
        TEST_TIME_AGO
      
      </span>
    </h4>
     
    <ul
      class="listing-icons"
    >
      <li>
        <i
          class="ln ln-icon-Management"
        />
         
        <a>
          
          TEST_NAME
        
        </a>
      </li>
       
      <li>
        <i
          class="ln ln-icon-Map2"
        />
         
        <a>
          
          TEST_LOCATION
        
        </a>
      </li>
       
      <li
        class="timeago-item"
      >
        <i
          class="ln ln-icon-Clock"
        />
         
        <span>
          TEST_TIME_AGO
        </span>
      </li>
       
      <li>
        <div
          class="listing-date post-tag tag-post-tag"
        >
          <a
            class="tag-post-tag"
          >
            
            TEST_SLUG_NAME_1
          
          </a>
        </div>
        <div
          class="listing-date post-tag tag-post-tag"
        >
          <a
            class="tag-post-tag"
          >
            
            TEST_SLUG_NAME_2
          
          </a>
        </div>
      </li>
    </ul>
  </div>
</a>
`;
